src/densities.jl

```diff
@@ -60,9 +60,10 @@ is not collinear the spin density is `nothing`.
     @assert n_k > 0
 
     # Allocate an accumulator for ρ in each thread for each spin component
-    ρaccus = [similar(view(ψ[1], :, 1), (basis.fft_size..., n_spin))
+    # TODO use something like T = promote_type(Array{eltype(basis)}, eltype(ψ), ...)
+    ρaccus = [similar(Array{complex(T)}, (basis.fft_size..., n_spin))
```

Comment:
Not just that ... this also makes it more annoying when we think about GPUs, but ok we worry about that some other day.

Reply:
That's a good point.. Is there a good alternative here?  Another solution would be to go for non-mutating code and avoiding the typed pre-allocation, which we'd explore next with Zygote anyways.

Reply:
I updated to use promote_type in combination with the three-arg version of `similar` specifying both the eltype and size https://github.com/JuliaMolSim/DFTK.jl/pull/476/commits/6a821bd79a920dafe7b2001db6609d8aba9644c3